File: frontend/src/utils/timeUtils.js
/**
 * 前端统一时间工具类
 * 解决前后端时间同步问题
 */

import dayjs from 'dayjs'
import utc from 'dayjs/plugin/utc'
import timezone from 'dayjs/plugin/timezone'
import relativeTime from 'dayjs/plugin/relativeTime'
import 'dayjs/locale/zh-cn'

// 扩展dayjs插件
dayjs.extend(utc)
dayjs.extend(timezone)
dayjs.extend(relativeTime)
dayjs.locale('zh-cn')

/**
 * 系统默认时区 - 中国标准时间
 */
const SYSTEM_TIMEZONE = 'Asia/Shanghai'

/**
 * 标准日期时间格式
 */
const DEFAULT_DATETIME_FORMAT = 'YYYY-MM-DD HH:mm:ss'

/**
 * 标准日期格式
 */
const DEFAULT_DATE_FORMAT = 'YYYY-MM-DD'

/**
 * ISO格式（用于前后端传输）
 */
const ISO_FORMAT = 'YYYY-MM-DDTHH:mm:ss.SSS[Z]'

export class TimeUtils {
  /**
   * 获取当前时间（系统时区）
   * 使用本地时间，如需服务器时间同步请使用时间同步服务
   * @returns {dayjs.Dayjs} 当前时间
   */
  static now() {
    return dayjs().tz(SYSTEM_TIMEZONE)
  }

  /**
   * 获取当前UTC时间
   * @returns {dayjs.Dayjs} 当前UTC时间
   */
  static utcNow() {
    return dayjs().utc()
  }

  /**
   * 获取当前日期
   * @returns {dayjs.Dayjs} 当前日期
   */
  static today() {
    return dayjs().tz(SYSTEM_TIMEZONE).startOf('day')
  }

  /**
   * 将服务器时间转换为本地显示时间
   * @param {string|Date|dayjs.Dayjs} serverTime 服务器时间
   * @param {string} format 格式化字符串
   * @returns {string} 格式化后的时间字符串
   */
  static formatServerTime(serverTime, format = DEFAULT_DATETIME_FORMAT) {
    if (!serverTime) return ''
    
    try {
      // 假设服务器时间是系统时区时间
      return dayjs(serverTime)
        .tz(SYSTEM_TIMEZONE)
        .format(format)
    } catch (error) {
      console.warn('时间格式化失败:', error)
      return ''
    }
  }

  /**
   * 格式化日期
   * @param {string|Date|dayjs.Dayjs} date 日期
   * @param {string} format 格式化字符串
   * @returns {string} 格式化后的日期字符串
   */
  static formatDate(date, format = DEFAULT_DATE_FORMAT) {
    if (!date) return ''
    
    try {
      return dayjs(date).tz(SYSTEM_TIMEZONE).format(format)
    } catch (error) {
      console.warn('日期格式化失败:', error)
      return ''
    }
  }

  /**
   * 格式化为ISO字符串（用于发送给后端）
   * @param {string|Date|dayjs.Dayjs} time 时间
   * @returns {string} ISO格式字符串
   */
  static formatToISO(time) {
    if (!time) return ''
    
    try {
      return dayjs(time).tz(SYSTEM_TIMEZONE).utc().format(ISO_FORMAT)
    } catch (error) {
      console.warn('ISO格式化失败:', error)
      return ''
    }
  }

  /**
   * 从ISO字符串解析时间
   * @param {string} isoString ISO格式字符串
   * @returns {dayjs.Dayjs|null} dayjs对象
   */
  static parseFromISO(isoString) {
    if (!isoString) return null
    
    try {
      return dayjs(isoString).utc().tz(SYSTEM_TIMEZONE)
    } catch (error) {
      console.warn('ISO解析失败:', error)
      return null
    }
  }

  /**
   * 获取相对时间描述
   * @param {string|Date|dayjs.Dayjs} time 时间
   * @returns {string} 相对时间描述
   */
  static fromNow(time) {
    if (!time) return ''
    
    try {
      const targetTime = dayjs(time).tz(SYSTEM_TIMEZONE)
      const now = this.now()
      const diffMinutes = now.diff(targetTime, 'minute')
      
      if (diffMinutes < 1) {
        return '刚刚'
      } else if (diffMinutes < 60) {
        return `${diffMinutes}分钟前`
      } else if (diffMinutes < 1440) { // 24小时
        const hours = Math.floor(diffMinutes / 60)
        return `${hours}小时前`
      } else if (diffMinutes < 10080) { // 7天
        const days = Math.floor(diffMinutes / 1440)
        return `${days}天前`
      } else {
        return this.formatServerTime(time)
      }
    } catch (error) {
      console.warn('相对时间计算失败:', error)
      return ''
    }
  }

  /**
   * 判断是否为同一天
   * @param {string|Date|dayjs.Dayjs} date1 日期1
   * @param {string|Date|dayjs.Dayjs} date2 日期2
   * @returns {boolean} 是否同一天
   */
  static isSameDate(date1, date2) {
    if (!date1 || !date2) return false
    
    try {
      return dayjs(date1).tz(SYSTEM_TIMEZONE).isSame(
        dayjs(date2).tz(SYSTEM_TIMEZONE), 'day'
      )
    } catch (error) {
      return false
    }
  }

  /**
   * 判断是否为今天
   * @param {string|Date|dayjs.Dayjs} date 日期
   * @returns {boolean} 是否为今天
   */
  static isToday(date) {
    return this.isSameDate(date, this.today())
  }

  /**
   * 获取时间戳（毫秒）
   * @param {string|Date|dayjs.Dayjs} time 时间
   * @returns {number} 时间戳
   */
  static toTimestamp(time) {
    if (!time) return 0
    
    try {
      return dayjs(time).tz(SYSTEM_TIMEZONE).valueOf()
    } catch (error) {
      return 0
    }
  }

  /**
   * 从时间戳创建时间对象
   * @param {number} timestamp 时间戳（毫秒）
   * @returns {dayjs.Dayjs} dayjs对象
   */
  static fromTimestamp(timestamp) {
    return dayjs(timestamp).tz(SYSTEM_TIMEZONE)
  }

  /**
   * 创建统一的当前时间对象（用于前端显示）
   * @returns {Object} 时间对象
   */
  static createCurrentTimeObject() {
    const now = this.now()
    return {
      timestamp: now.valueOf(),
      formatted: now.format(DEFAULT_DATETIME_FORMAT),
      iso: now.utc().format(ISO_FORMAT),
      relative: '刚刚'
    }
  }

  /**
   * 验证日期格式
   * @param {string} dateString 日期字符串
   * @param {string} format 期望格式
   * @returns {boolean} 是否有效
   */
  static isValidDate(dateString, format = DEFAULT_DATE_FORMAT) {
    if (!dateString) return false
    return dayjs(dateString, format, true).isValid()
  }

  /**
   * 获取日期范围
   * @param {number} days 天数
   * @returns {Object} 日期范围
   */
  static getDateRange(days = 30) {
    const today = this.today()
    const maxDate = today.add(days, 'day')
    
    return {
      min: today.toDate(),
      max: maxDate.toDate(),
      minFormatted: today.format(DEFAULT_DATE_FORMAT),
      maxFormatted: maxDate.format(DEFAULT_DATE_FORMAT)
    }
  }
}

// 导出常用格式常量
export const TIME_FORMATS = {
  DATETIME: DEFAULT_DATETIME_FORMAT,
  DATE: DEFAULT_DATE_FORMAT,
  TIME: 'HH:mm:ss',
  ISO: ISO_FORMAT,
  DISPLAY_DATETIME: 'MM月DD日 HH:mm',
  DISPLAY_DATE: 'MM月DD日'
}

// 导出时区常量
export const TIMEZONES = {
  SYSTEM: SYSTEM_TIMEZONE,
  UTC: 'UTC'
}

// 默认导出
export default TimeUtils

File: backend/src/main/java/com/carwash/utils/TimeUtils.java
package com.carwash.utils;

import java.time.*;
import java.time.format.DateTimeFormatter;
import java.util.TimeZone;

/**
 * 统一时间工具类
 * 解决前后端时间同步问题
 */
public class TimeUtils {
    
    /**
     * 系统默认时区 - 中国标准时间
     */
    public static final ZoneId SYSTEM_ZONE = ZoneId.of("Asia/Shanghai");
    
    /**
     * UTC时区
     */
    public static final ZoneId UTC_ZONE = ZoneId.of("UTC");
    
    /**
     * 标准日期时间格式
     */
    public static final DateTimeFormatter DEFAULT_DATETIME_FORMAT = 
        DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
    
    /**
     * 标准日期格式
     */
    public static final DateTimeFormatter DEFAULT_DATE_FORMAT = 
        DateTimeFormatter.ofPattern("yyyy-MM-dd");
    
    /**
     * ISO格式（用于前后端传输）
     */
    public static final DateTimeFormatter ISO_FORMAT = 
        DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
    
    /**
     * 获取当前系统时间
     * @return 当前LocalDateTime（系统时区）
     */
    public static LocalDateTime now() {
        return LocalDateTime.now(SYSTEM_ZONE);
    }
    
    /**
     * 获取当前UTC时间
     * @return 当前UTC时间
     */
    public static LocalDateTime utcNow() {
        return LocalDateTime.now(UTC_ZONE);
    }
    
    /**
     * 获取当前日期
     * @return 当前LocalDate（系统时区）
     */
    public static LocalDate today() {
        return LocalDate.now(SYSTEM_ZONE);
    }
    
    /**
     * 将LocalDateTime转换为时间戳（毫秒）
     * @param dateTime 本地时间
     * @return 时间戳
     */
    public static long toTimestamp(LocalDateTime dateTime) {
        if (dateTime == null) return 0L;
        return dateTime.atZone(SYSTEM_ZONE).toInstant().toEpochMilli();
    }
    
    /**
     * 将时间戳转换为LocalDateTime
     * @param timestamp 时间戳（毫秒）
     * @return LocalDateTime
     */
    public static LocalDateTime fromTimestamp(long timestamp) {
        return LocalDateTime.ofInstant(
            Instant.ofEpochMilli(timestamp), 
            SYSTEM_ZONE
        );
    }
    
    /**
     * 格式化时间为标准字符串
     * @param dateTime 时间
     * @return 格式化后的字符串
     */
    public static String formatDateTime(LocalDateTime dateTime) {
        if (dateTime == null) return "";
        return dateTime.format(DEFAULT_DATETIME_FORMAT);
    }
    
    /**
     * 格式化日期为标准字符串
     * @param date 日期
     * @return 格式化后的字符串
     */
    public static String formatDate(LocalDate date) {
        if (date == null) return "";
        return date.format(DEFAULT_DATE_FORMAT);
    }
    
    /**
     * 格式化时间为ISO字符串（用于前端传输）
     * @param dateTime 时间
     * @return ISO格式字符串
     */
    public static String formatToISO(LocalDateTime dateTime) {
        if (dateTime == null) return "";
        return dateTime.atZone(SYSTEM_ZONE)
                      .withZoneSameInstant(UTC_ZONE)
                      .format(ISO_FORMAT);
    }
    
    /**
     * 从ISO字符串解析时间
     * @param isoString ISO格式字符串
     * @return LocalDateTime
     */
    public static LocalDateTime parseFromISO(String isoString) {
        if (isoString == null || isoString.trim().isEmpty()) {
            return null;
        }
        try {
            return LocalDateTime.parse(isoString, ISO_FORMAT)
                               .atZone(UTC_ZONE)
                               .withZoneSameInstant(SYSTEM_ZONE)
                               .toLocalDateTime();
        } catch (Exception e) {
            return null;
        }
    }
    
    /**
     * 解析日期字符串
     * @param dateString 日期字符串
     * @return LocalDate
     */
    public static LocalDate parseDate(String dateString) {
        if (dateString == null || dateString.trim().isEmpty()) {
            return null;
        }
        try {
            return LocalDate.parse(dateString, DEFAULT_DATE_FORMAT);
        } catch (Exception e) {
            return null;
        }
    }
    
    /**
     * 解析日期时间字符串
     * @param dateTimeString 日期时间字符串
     * @return LocalDateTime
     */
    public static LocalDateTime parseDateTime(String dateTimeString) {
        if (dateTimeString == null || dateTimeString.trim().isEmpty()) {
            return null;
        }
        try {
            return LocalDateTime.parse(dateTimeString, DEFAULT_DATETIME_FORMAT);
        } catch (Exception e) {
            return null;
        }
    }
    
    /**
     * 判断是否为同一天
     * @param date1 日期1
     * @param date2 日期2
     * @return 是否同一天
     */
    public static boolean isSameDate(LocalDate date1, LocalDate date2) {
        if (date1 == null || date2 == null) return false;
        return date1.equals(date2);
    }
    
    /**
     * 判断是否为今天
     * @param date 要检查的日期
     * @return 是否为今天
     */
    public static boolean isToday(LocalDate date) {
        return isSameDate(date, today());
    }
    
    /**
     * 获取相对时间描述
     * @param dateTime 时间
     * @return 相对时间描述
     */
    public static String getRelativeTime(LocalDateTime dateTime) {
        if (dateTime == null) return "";
        
        LocalDateTime now = now();
        Duration duration = Duration.between(dateTime, now);
        
        long seconds = duration.getSeconds();
        long minutes = seconds / 60;
        long hours = minutes / 60;
        long days = hours / 24;
        
        if (seconds < 60) {
            return "刚刚";
        } else if (minutes < 60) {
            return minutes + "分钟前";
        } else if (hours < 24) {
            return hours + "小时前";
        } else if (days < 7) {
            return days + "天前";
        } else {
            return formatDateTime(dateTime);
        }
    }
    
    /**
     * 设置系统时区
     */
    public static void setSystemTimeZone() {
        TimeZone.setDefault(TimeZone.getTimeZone(SYSTEM_ZONE));
        System.setProperty("user.timezone", SYSTEM_ZONE.getId());
    }
}